README.md

```diff
@@ -1,14 +1,62 @@
+
+## The Rules
+The game starts off with the position as shown in the image. Generally, the red pieces will get to move first. The game ends when either sides have no more moves to save the general piece from being taken in the next move, similar to checkmating the king in chess. 
+
+When a player make a move which puts the enemy general in danger of being capture in the next move, that is called a check. Unlike chess, perpetual checking of the enemy king results not in stalemate, but disqualification of the player making the perpetual check. Specifically, the same piece cannot be used to check the enemy general in the same position for more than 3 times in a row.
+
+There is also a rule where perpetually chasing a piece is not allowed. In our implementation, this is not added as we believe that stopping the process of perpetually chasing a piece should be a part of the strategy.
+
+## The Pieces
+Each player starts off with 1 general, 2 advisors, 2 elephants, 2 horses, 2 chariots, 2 cannons and 5 soldiers.
+
+### The General
+The general lives in the palace, and can only move within the palace. The general can move orthogonally 1 position, which means that it can only move forward, backward, left or right by 1 position. The general can be used to capture a piece as it moves, unless capturing a piece puts the general in check.
+
+A special rule called the "flying general" applies to the general, where if the two generals from opposing side faces each other in an open (no pieces blocking) vertical column, the general can "fly" and capture the opponent's general. Since a general will occupy an open vertical column first, the opponent will not be able to move their general into the open column since that will be a movement into check. However, this rule can be used to checkmate an opponent's general.
+
+### The Advisor
+Similar to the general, the advisor can only move within the palace. In contrast to the general, the advisor can only move diagonally, which means that there are only 5 possible positions that each advisor can take. The advisor can be used to capture a piece as it moves.
+
+### The Elephant
+The elephant moves exactly 2 steps diagonally, which can also be visualized as two positions to the left/right and two positions forward/backward. The elephant will not be able to jump over pieces while moving, meaning if a piece exists to the top right of the elephant, it will not be able to jump over that piece to reach the top right position of the piece.
+
+The elephant can be used to a piece as it moves. The elephant cannot cross the river, which is why it is mainly used as a defensive piece.
```

Comment:
The first sentence seems to have a word missing or typo.

Reply:
Done